### DISCUSSION TITLE
Make `i32` the concrete type and `int` the alias

### DISCUSSION THREAD

**Discussion by txgruppi** [vlang/v#14622] [Ideas]
It feels wrong that we have the integer type sequece as `i8`, `i16`, `int`, `i64` and even worse if we consider `u8`, `u16`, `u32`, `u64`.

I propose we have `i32` the concrete type so we keep the naming consistent and have `int` as an alias to `i32`.

**Comment by JalonSolov** [comment]
It has been discussed, just hasn't happened yet.

**Comment by king8fisher** [comment]
Taking a note on this in conjunction with `in` and `is`:

As with normal `if` expressions, you can use the `in` statement:

```v nofmt failcompile
$if field.typ in [string, int] {
	println('${field.name} is of type string or int')
}
```

> **Note**
> Although `i32` is defined as a `type alias i32 = int`, they are treated as separate types. Therefore, a field of type `i32` will **not** match either `is int` or `in [int]`. The same holds true in reverse.
